test/test_intentstore_interface.py

```diff
@@ -0,0 +1,73 @@
+# -*- coding: utf-8 -*-
+
+from __future__ import print_function, division, absolute_import
+
+#
+# Copyright (c) 2018 Red Hat, Inc.
+#
+# This software is licensed to you under the GNU General Public License,
+# version 2 (GPLv2). There is NO WARRANTY for this software, express or
+# implied, including the implied warranties of MERCHANTABILITY or FITNESS
+# FOR A PARTICULAR PURPOSE. You should have received a copy of GPLv2
+# along with this software; if not, see
+# http://www.gnu.org/licenses/old-licenses/gpl-2.0.txt.
+#
+# Red Hat trademarks are not licensed under GPLv2. No permission is
+# granted to use or replicate Red Hat trademarks that are incorporated
+# in this software or its documentation.
+#
+
+import json
+import os
+import shutil
+import tempfile
+import unittest
+
+import sys
+
+from subscription_manager import intentstore_interface
+from intentctl import intentfiles
+
+
+class IntentStoreInterfaceTests(unittest.TestCase):
+
+    def setUp(self):
+        temp_dir = tempfile.mkdtemp()
+        self.addCleanup(shutil.rmtree, temp_dir)
+        mock_intent_file = os.path.join(temp_dir, "mock_intent.json")
+        intent_values = {}
+        with open(mock_intent_file, 'w') as f:
+            json.dump(intent_values, f)
+            f.flush()
+        intentfiles.USER_INTENT = mock_intent_file
+
+    def tearDown(self):
+        intentfiles.USER_INTENT = "/etc/rhsm/intent/intent.json"
+
+    def test_set_intent_service_level_sla_is_set_when_intentctl_module_exists(self):
+        """
+        Tests that the intent sla is set through the intentstore interface
+        when the intentctl module is available for import.
+        """
+        intentstore_interface.set_intent_service_level("Freemium")
+
+        contents = intentfiles.IntentStore.read(intentfiles.USER_INTENT).contents
+        self.assertEqual(contents.get("service_level_agreement"), "Freemium")
+
```

Comment:
Please add test function for following case: intentstore_interface.set_intent_service_level(None). File `intent.json` should contain line: `"service_level_agreement": null`